### PR TITLE
Protocol buffer basic CMake for static lib added

### PR DESCRIPTION
**Pull request by AakashMallik**
**Reference issue**  
#12 

**PR description:**
Added a sample protobuf for things. Planning to use this inside a wrapper class.

**List of changes made in this PR:**
- Added a proto folder
- Added a new cmake
- Updated parent cmake

**Does this break the platform?**
Build breaking as of now

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -5,6 +5,9 @@
 // lib
 #include "crow_all.h"
 
+// proto
+#include "thing_buffer.h"
```
Comment (AakashMallik) [inline src/main.cpp]: Make unable to find this... @dhirajfx3

File: .vscode/c_cpp_properties.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+            ],
+            "defines": [],
+            "compilerPath": "/usr/bin/gcc",
+            "cStandard": "c11",
+            "cppStandard": "c++17",
+            "intelliSenseMode": "clang-x64"
+        }
+    ],
+    "version": 4
+}
```
Comment (AakashMallik) [inline .vscode/c_cpp_properties.json]: Ignore this

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
 	"files.associations": {
 		"vector": "cpp"
-	}
+	},
 }
```
Comment (AakashMallik) [inline .vscode/settings.json]: Ignore this...

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -27,15 +27,15 @@ endforeach(DIR)
...
 # create executable
 add_executable(${PROJECT_NAME} ${SRC_PATHS} ./main.cpp)
 
-# linking executable with libraries
+# linking executable with required libraries
 target_link_libraries(
-  ${PROJECT_NAME} 
-  ${LIBRARY_LINK_PATHS}
+	${PROJECT_NAME} 
+	protos
```
Comment (dhirajfx3) [inline src/CMakeLists.txt]: target_link_libraries( ${PROJECT_NAME} LINK_PUBLIC protos ${BOOST_LIBRARIES} )

File: src/main.cpp [hunk 5]
```diff
@@ -5,6 +5,9 @@
 // lib
 #include "crow_all.h"
 
+// proto
+#include "thing_buffer.bh.h"
```
Comment (dhirajfx3) [inline src/main.cpp]: pb.h

**Comment by AakashMallik** [comment]
No clue why isn't it able to find the header.

**Pull request merged**